foreign_cc/private/framework.bzl

```diff
@@ -857,3 +866,30 @@ def _collect_libs(cc_linking):
                 if library:
                     libs.append(library)
     return collections.uniq(libs)
+
+def _generate_make_commands(ctx):
+    make_commands = getattr(ctx.attr, "make_commands", [])
+    tools_deps = []
+
+    # Early out if there are no commands set
+    if not make_commands:
+        return make_commands, tools_deps
+
+    if _uses_tool(ctx.attr.make_commands, "make"):
+        make_data = get_make_data(ctx)
+        tools_deps += make_data.deps
+        make_commands = [command.replace("make", make_data.path) for command in make_commands]
```

Comment:
Additionally, this change seems wrong to me. Before, the logic was:
``` python
if line == "make" or line.startswith("make "):
    make_commands.append(line.replace("make", attrs.make_path, 1))
```

This will now change something like `["make", "class_maker"]` to `["/usr/bin/make", "class_/usr/bin/maker"]`.